packages/utils/src/data.ts

```diff
@@ -22,19 +22,19 @@ export function decodeBool(encodedBool: Uint8Array): boolean {
   return false;
 }
 
-export function encodeInt(int: number | bigint): Uint8Array {
-  return bigIntToScriptNumber(BigInt(int));
+export function encodeInt(int: bigint): Uint8Array {
+  return bigIntToScriptNumber(int);
```

Comment:
`encodeInt` is now just a somewhat less clear name for `bigIntToScriptNumber` so not sure how useful it is to have this wrapper still...

Reply:
I think this is a very good point, but I also think that it's best to keep it as-is for now for consistency with the other encode/decode functions. Let's revisit this in the future if necessary.